mgr/dashboard: add smb share listing in UI

Pegonzal opened the pull request:
Fixes: https://tracker.ceph.com/issues/69449

[screen-capture (44).webm](https://github.com/user-attachments/assets/d2921945-7545-4d91-bc12-0e53948278ea)


<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

Pegonzal commented:
jenkins test make check

dnyanee1997 commented:
@Pegonzal ,
I am testing the PR and found that when i expand the cluster it's getting closed after few seconds.
Here is the attached recording. Is it the expected behavior

https://github.com/user-attachments/assets/2193fdf8-00e1-4e99-940b-7064098806ac

Pegonzal commented:
> @Pegonzal , I am testing the PR and found that when i expand the cluster it's getting closed after few seconds. Here is the attached recording. Is it the expected behavior?
> 
> Seems like page is not stable
> 
>  Screencast.from.2025-01-15.17-25-40.mp4

Thanks for testing it @dnyanee1997, I had missed that before, now it should be fixed and working as expected.
I've attached a video in the PR description

Pegonzal commented:
> Hi @Pegonzal, Changes looks good to me, but if possible can we start using carbon tabs in the new development.

Good idea, thanks @nmunet for the suggestion, I'll look into the carbon tabs

dnyanee1997 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-list/smb-share-list.component.ts:
```diff
@@ -0,0 +1,89 @@
+  table: TableComponent;
+  columns: CdTableColumn[];
+  permission: Permission;
+  tableActions: CdTableAction[];
```
```suggestion
```

nizamial09 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-list/smb-share-list.component.ts:
```diff
@@ -0,0 +1,89 @@
+        flexGrow: 2
+      },
+      {
+        name: $localize`Path`,
```
We do have a path template that we use (in subvolume and subvol group tables), where it'll trim if its a long path and allows you to copy that path as well. Shall we use it here too?

Pegonzal replied:
Updated to using celltemplate path now

nizamial09 reviewed (approved):
Thanks @Pegonzal

nizamial09 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-tabs/smb-cluster-tabs.component.html:
```diff
@@ -0,0 +1,21 @@
+  <nav
+    ngbNav
+    #nav="ngbNav"
+    class="nav-tabs"
```
shouldn't we move to [carbon-tabs?](https://carbondesignsystem.com/components/tabs/code/)

Pegonzal replied:
Yes, I have updated the PR using carbon tags now. Also updated the video in PR description

Pegonzal commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-list/smb-share-list.component.ts:
```diff
@@ -0,0 +1,89 @@
+        flexGrow: 2
+      },
+      {
+        name: $localize`Subvolumegroup`,
```
Yep, and updated it, thanks

Pegonzal commented:
Thanks for the reviews @nmunet ,  @dnyanee1997 , @nizamial09, I have updated the PR and the video in the PR description. Please take another look

dnyanee1997 commented:
LGTM.Thanks @Pegonzal

Pegonzal commented:
jenkins test make check arm64

Pegonzal commented:
jenkins test api

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-tabs/smb-cluster-tabs.component.spec.ts:
```diff
@@ -0,0 +1,22 @@
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
we can add a test for tab heading

Pegonzal replied:
I have added a couple tests verifying the selected item and the rendering of the tabs.
I am not sure if any other tests are expected, let me know if you find anything else worth to add as I wasn't really sure what to unit test from here.

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/styles/_carbon-defaults.scss:
```diff
@@ -149,3 +149,10 @@ Code snippet
+Tabs
+******************************************/
+.cds--tab-content {
+  background: var(--cds-layer-01);
```
why this ?

Pegonzal replied:
Hey @afreen23, the angular contained tabs implementation does not fill the background color to match the active tab color. So this takes care of it, by coloring the contined space of the tab.

If you take a look at [React's implementation of this feature](https://react.carbondesignsystem.com/?path=/story/components-tabs--contained): you can see the gray contained space.

Does that no happen in the [Angular's implementation of this feature](https://angular.carbondesignsystem.com/?path=/story/components-tabs--basic&args=type:contained): hence this adds this gray coloring

I though it looked better this way, let me know your thoughts

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb.model.ts:
```diff
@@ -26,3 +26,29 @@ export interface AuthMode {
+  cluster_id: string;
+  share_id: string;
+  intent: string;
+  name?: string;
+  readonly?: boolean;
+  browseable?: boolean;
```
nit: move optional props down

Pegonzal replied:
It was initially in the same order as in the docs, I've updated it to have optional fields last

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb.model.ts:
```diff
@@ -26,3 +26,29 @@ export interface AuthMode {
+
+interface SMBShareLoginControl {
+  name: string;
+  category?: 'user' | 'group';
```
move optional down

Pegonzal replied:
It was initially in the same order as in the docs, I've updated it to have optional fields last

Pegonzal commented:
@afreen23 Thanks afreen for the review, I've updated the PR after your requested changes, please have another look

Pegonzal commented:
jenkins test make check

Pegonzal commented:
jenkins test api

afreen23 reviewed (changes requested):
There seems soem space betweed shares tab and the table.PTAL
![image](https://github.com/user-attachments/assets/9af38436-8c35-4962-85c9-3f7fd1c70f4e)

Thanks

Pegonzal commented:
> There seems soem space betweed shares tab and the table.PTAL ![image](https://private-user-images.githubusercontent.com/25664409/407228501-9af38436-8c35-4962-85c9-3f7fd1c70f4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwNTIxMzIsIm5iZiI6MTczODA1MTgzMiwicGF0aCI6Ii8yNTY2NDQwOS80MDcyMjg1MDEtOWFmMzg0MzYtOGMzNS00OTYyLTg1YzktM2Y3ZmQxYzcwZjRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDA4MTAzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxNGY1NjhjNjZkNTczNmIyMmMxYWExMzU0ZmE1ZTAwOGM0OTJkODRkZWE0YzI5MjA2MzNkOWJlZWRmYjUyOTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nj31ohOaa5rT3M2MZ5HdsKB34CYLswW9zbwuO5l-ZmQ)
> 
> Thanks

Not sure what you are refferring to @afreen23 .
What should I change?

Pegonzal commented:
@afreen23 I've updated the PR. 

Let me know if now looks as expected

[screen-capture (47).webm](https://github.com/user-attachments/assets/a69c921f-4e9f-4ecf-b7cf-abc7756c89b9)

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-list/smb-share-list.component.ts:
```diff
@@ -0,0 +1,89 @@
+        flexGrow: 2
+      },
+      {
+        name: $localize`File System`,
+        prop: 'cephfs.volume',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Path`,
+        prop: 'cephfs.path',
+        cellTransformation: CellTemplate.path,
+        flexGrow: 2
+      },
+      {
+        name: $localize`Subvolume group`,
+        prop: 'cephfs.subvolumegroup',
+        flexGrow: 2
+      },
```
Either we use camel case or not. Lets chekc with @bhkaur25 @rivkap

Pegonzal replied:
Yes, good point. I went with: `File System` because we have it like that in other places and same thing for `Subvolume group` that's how is written in the cephfs components.

We can look to unify all these in another PR

afreen23 commented:
> @afreen23 I've updated the PR.
> 
> Let me know if now looks as expected
> 
>  screen-capture.47.webm

Nice! I think the only issue is color we dont want carbon default blue, and try to replicate look and feel as we were trying out in this PR https://github.com/ceph/ceph/pull/60349/ for contained tabs.
Thanks

Pegonzal commented:
> > @afreen23 I've updated the PR.
> > Let me know if now looks as expected
> > screen-capture.47.webm
> 
> Nice! I think the only issue is color we dont want carbon default blue, and try to replicate look and feel as we were trying out in this PR #60349 for contained tabs. Thanks

I've added the border-primery color. @afreen23 
I think we that all the requested changes are made.

Current look and feel:
![image](https://github.com/user-attachments/assets/9bed4482-d93e-44f0-b87a-6e7bc1cbea9c)

afreen23 commented on src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-list/smb-share-list.component.ts:
```diff
@@ -0,0 +1,89 @@
+        flexGrow: 2
+      },
+      {
+        name: $localize`File System`,
```
```suggestion
        name: $localize`FileSystem`,
```

afreen23 replied:
I think it should be this, PTAL cc @bhkaur25

Pegonzal replied:
Regardless, I went with `File System` as it is how it is displayed in the left-side menu as well as the `File System` page:
![image](https://github.com/user-attachments/assets/c76ad7c5-fcfe-4a3c-9442-0f18829dbc53)
![image](https://github.com/user-attachments/assets/20458e10-b0ad-4fc4-9985-5d36b1a84666)

This is consistent through the application

afreen23 reviewed (changes requested):
LGTM, slightly off right margin in table if you can fix that it would be nice, since I am not seeing that other places in dashboard.
But great work from last time 👍🏿 
![image](https://github.com/user-attachments/assets/78ce84f1-2868-4693-9fef-16bfaaca421e)

Thanks @Pegonzal

Pegonzal commented:
> LGTM, slightly off right margin in table if you can fix that it would be nice, since I am not seeing that other places in dashboard. But great work from last time 👍🏿 ![image](https://private-user-images.githubusercontent.com/25664409/409453612-78ce84f1-2868-4693-9fef-16bfaaca421e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2NjEwNjIsIm5iZiI6MTczODY2MDc2MiwicGF0aCI6Ii8yNTY2NDQwOS80MDk0NTM2MTItNzhjZTg0ZjEtMjg2OC00NjkzLTlmZWYtMTZiZmFhY2E0MjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA0VDA5MTkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkZWEyZmUwNWM1NTNmOTlkZTFiNTY1YjFmYWM1NzA2OGVhZGUwMGM4Y2U3N2MwM2QxMmE5Yjk4ODVlNDIzMjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n78E4boLGHHOsQpPJytTqnqT4o2lbvMtb_kDQ6z9R68)
> 
> Thanks @Pegonzal

Hey @afreen23, I think this is the same margin as other implementations, see:

![image](https://github.com/user-attachments/assets/ddc0d3f6-b67e-4cfa-9d5f-23680f665aaf)
From: https://github.com/ceph/ceph/pull/60349

![image](https://github.com/user-attachments/assets/018df189-8601-4f8b-8058-f2280089c268)
From: https://github.com/ceph/ceph/pull/61273

I believe this default margin follows carbon guidelines, the tab's text aligns with the content below
![image](https://github.com/user-attachments/assets/2cb9af98-acc5-4a9a-8a85-0bde6fd84ef4)
From: https://carbondesignsystem.com/components/tabs/usage/

In this case, the content below is a table, and as you highlited it aligns with it:
![image](https://github.com/user-attachments/assets/bcdaa1d2-aec2-4e60-b8f7-fab40572bb8a)


Hence, I am not entirely sure of what the requested change is.
Could you elaborate? Thanks.

Pegonzal commented:
jenkins test make check arm64

Pegonzal commented:
jenkins test make check arm64

Pegonzal commented:
jenkins test make check arm64
